server/src/tenant-datasource/tenant-datasource.service.ts

```diff
@@ -22,21 +22,33 @@ export class TenantDataSourceService {
   public async connectToWorkspaceDataSource(
     workspaceId: string,
   ): Promise<DataSource> {
-    const dataSourceMetadata =
+    const { id, schema } =
       await this.dataSourceService.getLastDataSourceMetadataFromWorkspaceIdOrFail(
         workspaceId,
       );
+    return this.connectToSchemaDataSource(schema, id);
+  }
 
-    const dataSource = await this.typeormService.connectToDataSource(
-      dataSourceMetadata,
-    );
-
+  /**
+   *
+   * Connect to the schema data source
+   *
+   * @params schema, dataSourceId
+   * @returns
+   */
+  public async connectToSchemaDataSource(
+    schema: string,
+    dataSourceId?: string,
+  ): Promise<DataSource> {
+    const dataSource = await this.typeormService.connectToDataSource({
```

Comment:
Why do we want to deconstruct DataSource here? I'm not sure I understand this extra method.

Reply:
Otherwise we can also remove DataSource dependency from typeORM service and simply send 2 parameters which would be better than a Partial I think.